src/main/java/net/glowstone/block/blocktype/BlockType.java

```diff
@@ -303,11 +305,28 @@ public final void rightClickBlock(GlowPlayer player, GlowBlock against, BlockFac
             return;
         }
 
-        // TODO: check bounding box
         if (getMaterial().isSolid()) {
-            for (Entity entity : against.getChunk().getEntities()) {
-                if (entity instanceof LivingEntity && entity.getLocation().getBlockX() == target.getX() && entity.getLocation().getBlockY() == target.getY() && entity.getLocation().getBlockZ() == target.getZ()) {
-                    return;
+            GlowChunk[] chunks = new GlowChunk[]{
+                    target.getChunk(),
+                    target.getWorld().getChunkAt(target.getChunk().getX() + 1, target.getChunk().getZ()),
+                    target.getWorld().getChunkAt(target.getChunk().getX() + 1, target.getChunk().getZ() + 1),
+                    target.getWorld().getChunkAt(target.getChunk().getX(), target.getChunk().getZ() + 1),
+                    target.getWorld().getChunkAt(target.getChunk().getX() - 1, target.getChunk().getZ()),
+                    target.getWorld().getChunkAt(target.getChunk().getX() - 1, target.getChunk().getZ() - 1),
+                    target.getWorld().getChunkAt(target.getChunk().getX() - 1, target.getChunk().getZ() + 1),
+                    target.getWorld().getChunkAt(target.getChunk().getX(), target.getChunk().getZ() - 1),
+                    target.getWorld().getChunkAt(target.getChunk().getX() + 1, target.getChunk().getZ() - 1),
+            };
+            BlockBoundingBox box = new BlockBoundingBox(target);
+            for (GlowChunk chunk : chunks) {
+                for (Entity e : chunk.getEntities()) {
+                    GlowEntity entity = (GlowEntity) e;
+                    if (!(entity instanceof LivingEntity)) {
```

Comment:
Not really because the `return;` statement says that the block can't be placed (if an entity intersects the block), and you can place blocks on non-living entities (which is why there's a `continue;` statement.)